fe/src/main/java/org/apache/doris/analysis/FromClause.java

```diff
@@ -72,6 +78,39 @@ public int compare(TableRef tableref1, TableRef tableref2) {
         });
     }
 
+    private void checkFromHiveTable(Analyzer analyzer) throws AnalysisException {
+        for (TableRef tblRef : tableRefs_) {
+            if (!(tblRef instanceof BaseTableRef)) {
+                continue;
+            }
+
+            TableName tableName = tblRef.getName();
+            String dbName = tableName.getDb();
+            if (Strings.isNullOrEmpty(dbName)) {
+                dbName = analyzer.getDefaultDb();
+            } else {
+                dbName = ClusterNamespace.getFullName(analyzer.getClusterName(), tblRef.getName().getDb());
+            }
+            if (Strings.isNullOrEmpty(dbName)) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+
+            Database db = analyzer.getCatalog().getDb(dbName);
+            if (db == null) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_DB_ERROR, dbName);
+            }
+
+            String tblName = tableName.getTbl();
+            Table table = db.getTable(tblName);
```

Comment:
Lock db before getting table

Reply:
Lock is acquired in StmtExecutor.analyze function.